### ISSUE TITLE
Help: Selecting compatible rotary encoder

### ISSUE THREAD

**Issue by gorjious** [josefadamcik/SofleKeyboard#89]
**Kind of issue?**
Assembly / Software

**Which Board?**
RGB  

**Which Board Revision?**
Not certain. It is the Sofle RGB from Keyhive
https://keyhive.xyz/shop/sofle

**What is the issue?**
I recently purchased a Sofle RGB (will be my first build) and the rotary encoder it came with has detents. I would like to be able to select a new rotary encoder that will have a smooth endless rotation. There are so many options with rotary encoder types that I would like to ask for help in selecting which type will be compatible with the board and the software. 

On your website, the recommendation is for EC11. Will all these options in the EC11 series be compatible with the Sofle? https://tech.alpsalpine.com/prod/e/html/encoder/incremental/ec11/ec11_list.html

However, something like this [Dual Rotary Encoder](https://www.aliexpress.com/item/33052809089.html?albpd=en33052809089&acnt=708-803-3821&aff_platform=aaf&albpg=896505316494&netw=u&albcp=9824643245&sk=UneMJZVf&trgt=896505316494&terminal_id=91dceef5c23940688d33610728253fab&tmLog=new_Detail&needSmbHouyi=false&albbt=Google_7_shopping&src=google&crea=en33052809089&aff_fcid=0acda23e23f74e42ad41aa9dde35df49-1623774908644-06289-UneMJZVf&gclid=CjwKCAjwn6GGBhADEiwAruUcKr4Vffws_OgxEDryxWltp7F9-KviPHxddABNW6xUpdlkCFdgRuTCxxoCcwwQAvD_BwE&albag=103112554314&aff_fsk=UneMJZVf&albch=shopping&albagn=888888&isSmbAutoCall=false&aff_trace_key=0acda23e23f74e42ad41aa9dde35df49-1623774908644-06289-UneMJZVf&device=c&gclsrc=aw.ds) is also part of the EC11 series but it has extra pins and doesn't seem like it would be compatible with the current board.

What are the factors that I should keep in mind when selecting an encoder to work with the board and with QMK firmware? What should be avoided?
Is there a difference in selecting an encoder with a higher amount of detents (36 detents) or detentless? 

Any guidance or recommendations on specific encoders will be greatly appreciated.

**Comment by josefadamcik** [comment]
TL;DR; the footprint/pinout is the important thing. If you want to be sure, you'll have to check if the encoder you have chosen matches the footprint.

> On your website, the recommendation is for EC11

I am aware it's not really accurate specification, sorry for that. My own experiments were limited to few encoders from aliexpress and I never had a real part number for any of them. On the other hand, most of the people wouldn't go for exact part number anyway.  It's still a bit better than write "an encoder with matching footprint" :D

> Will all these options in the EC11 series be compatible with the Sofle? https://tech.alpsalpine.com/prod/e/html/encoder/incremental/ec11/ec11_list.html

I haven't checked the datasheets and haven't tested all the encoders in the list, so I cannot answer the question, sorry.

> However, something like this Dual Rotary Encoder is also part of the EC11 series but it has extra pins and doesn't seem like it would be compatible with the current board.

That is correct, this is certainly not compatible. 

> What are the factors that I should keep in mind when selecting an encoder to work with the board and with QMK? What should be avoided?

Given my limited knowledge of encoders and QMK I am not able to answer this question.

> Is there a difference in selecting an encoder with a higher amount of detents (36 detents) or detentless?

As long as the encoder sends digital pulses (which should be the case for all of them, I hope) I would guess not,  there isn't . 

> Any guidance or recommendations on specific encoders will be greatly appreciated.

Sorry, I don't even remember where I got the ones I am using.

In general consider asking somewhere where you would find more members of the split keyboard community (QMK discord corresponding reddit etc). You'll find a lot of people who experimented with various encoders and might have done their research an have more knowledge. Here you are reaching out to maybe 2 maintainers of the repository and few other people who might be watching. Also the seller of the keyboard surely has some support and/or discord. Since you are their customer, they should be thrilled to help you.





 I don'r really know how the precisely specify and encod
